libs/workspace-manager/src/lib/features/filesystem-navigator/filesystem-navigator.component.ts

```diff
@@ -0,0 +1,97 @@
+import { AsyncPipe, NgForOf, NgIf } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  inject,
+  Input,
+  OnChanges,
+  Output,
+} from '@angular/core';
+import { MatDividerModule } from '@angular/material/divider';
+import { MatIconModule } from '@angular/material/icon';
+import { MatListModule } from '@angular/material/list';
+import { Directory } from '@angular-cli-gui/shared/data';
+import { BehaviorSubject, Observable, of, switchMap, take } from 'rxjs';
+
+import { WorkspaceManagerApiService } from '../../data-access/workspace-manager-api.service';
+
+import { FilesystemNavigatorToolbarComponent } from './filesystem-navigator-toolbar/filesystem-navigator-toolbar.component';
+import { FilesystemNavigatorUtilsService } from './services/filesystem-navigator-utils.service';
+
+@Component({
+  selector: 'cli-filesystem-navigator',
+  standalone: true,
+  imports: [
+    MatListModule,
+    NgForOf,
+    NgIf,
+    MatDividerModule,
+    MatIconModule,
+    FilesystemNavigatorToolbarComponent,
+    AsyncPipe,
+  ],
+  templateUrl: './filesystem-navigator.component.html',
+  styleUrls: ['./filesystem-navigator.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FilesystemNavigatorComponent implements OnChanges {
+  private readonly fsNavigatorUtilsService = inject(
+    FilesystemNavigatorUtilsService
+  );
+  private readonly workspaceManagerApiService = inject(
+    WorkspaceManagerApiService
+  );
+
+  @Input() path: string | null = null;
+
+  @Output() pathChange = new EventEmitter<string>();
+
+  private readonly _isAngularWorkspace$ = new BehaviorSubject<boolean>(false);
+  readonly isAngularWorkspace$ = this._isAngularWorkspace$.asObservable();
+  directories$: Observable<Directory[]> = of([]);
+
+  ngOnChanges(): void {
+    if (this.path) {
+      this.directories$ = this.workspaceManagerApiService.getDirectoriesInPath(
+        this.path
+      );
+    }
+  }
+
+  onPathChanged(path: string): void {
+    this.directories$
+      .pipe(
+        take(1),
+        switchMap((directories: Directory[]) => {
+          const dirname =
+            this.fsNavigatorUtilsService.extractDirNameFromPath(path);
+          const directory = directories.find((dir) => dir.name === dirname);
+          return directory
+            ? of(directory)
+            : this.fsNavigatorUtilsService.getDirectoryFromParentPath(
```

Comment:
I don't think this component should make HTTP requests, I think it should be "stupider", and the component that uses it should send it the directories with input

Reply:
so this is the approach I started with, but then i thought - wouldn't it be better to use a simpler API, just provide a path and the fs-navigator will take care of the rest. 
don't forget we will have a lot of events to take care of like refresh, this means a lot of event handling for each consumer of this component.

Reply:
@ronnetzer @agroupp thoughts?

Reply:
Currently it's kinda both dumb and smart component, in general choose one of them and go with it. 
In this case I agree with @OmerGronich, lets make it dumb + service: don't write the logic of it in the consumer (like refresh and any other event) so it'll be easier to reuse it with other consumers